tensorflow/lite/delegates/webnn/webnn_delegate.cc

```diff
@@ -350,37 +337,46 @@ class Subgraph {
     }
 
     if (any_pointers_changed) {
-      graph_inputs_ = wnn::CreateNamedInputs();
+      graph_inputs_ = emscripten::val::object();
       for (int t : inputs_) {
-        wnn_inputs_[t].resource.arrayBufferView.buffer = context->tensors[t].data.raw;
-        wnn_inputs_[t].resource.arrayBufferView.byteLength = context->tensors[t].bytes;
         std::string name = std::to_string(t);
-        graph_inputs_.Set(name.c_str(), &wnn_inputs_[t]);
+        auto input_size = context->tensors[t].bytes / 4;
+        auto input_data = context->tensors[t].data.f;
+        emscripten::val view{ emscripten::typed_memory_view(input_size, input_data) };
+        auto input = emscripten::val::global("Float32Array").new_(input_size);
+        input.call<void>("set", view);
+        graph_inputs_.set(name, input);
```

Comment:
I suppose you could set `view` directly to `graph_inputs_` instead of copying its content to another `Float32Array`. Because the type of `input_data` should be `float`, according to [embind doc](https://emscripten.org/docs/porting/connecting_cpp_and_javascript/embind.html#memory-views), the view should be a `Float32Array` object in JS. Is my understanding right?

```js
graph_inputs_.set(name, view)
```

Reply:
Thanks much @huningxin ! You are correct, `view` should be typeof an `ArrayBuffer`. I've updated the code, PTAL again. Thanks!